samples/matter/common/src/board/board.cpp

```diff
@@ -293,8 +294,12 @@ void Board::DefaultMatterEventHandler(const ChipDeviceEvent *event, intptr_t /*
 
 	switch (event->Type) {
 	case DeviceEventType::kCHIPoBLEAdvertisingChange:
-		if (ConnectivityMgr().NumBLEConnections() != 0) {
+		if (ConnectivityMgr().NumBLEConnections() != 0 && !isNetworkProvisioned) {
 			sInstance.UpdateDeviceState(DeviceState::DeviceConnectedBLE);
+		} else if (!isNetworkProvisioned) {
+			sInstance.UpdateDeviceState(DeviceState::DeviceAdvertisingBLE);
+		} else {
```

Comment:
nit:

You could handle this as the first `if (isNetworkProvisioned)` and then you woudn't need to negate it in all previous conditions :)